samples/bluetooth/mesh/dfu/target/prj.conf

```diff
@@ -65,6 +65,17 @@ CONFIG_BT_MESH_DFU_SRV=y
 CONFIG_BT_MESH_DFU_METADATA=y
 CONFIG_BT_MESH_SAR_CFG_SRV=y
 
+# Enable DFU Server shell commands to let user cancel DFU from shell
+CONFIG_BT_MESH_SHELL=y
+CONFIG_BT_MESH_SHELL_DFU_SRV=y
+# Disable other shell commands that get enabled automatically
+CONFIG_BT_MESH_SHELL_BLOB_IO_FLASH=n
+CONFIG_BT_MESH_SHELL_BLOB_SRV=n
+CONFIG_BT_MESH_SHELL_DFU_METADATA=n
+CONFIG_BT_MESH_SHELL_GATT_PROXY=n
+CONFIG_BT_MESH_SHELL_TEST=n
+CONFIG_BT_MESH_SHELL_PROV=n
```

Comment:
Check if we want to disable `CONFIG_FLASH_SHELL` as well, if both `CONFIG_FLASH` and `CONFIG_SHELL` is enabled implicitly after your change. It consumes 8k of RAM and we don't need it.